boards/FSM/Inc/Io/Io_FlowMeters.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <stm32f3xx_hal.h>
+
+/**
+ * Initializes the primary and secondary flow meter
+ * @param htim : The timer handle for both flow meters
+ */
+void Io_FlowMeters_Init(TIM_HandleTypeDef *htim);
+
+/**
+ * Get flow rate (L/min) from the primary flow meter
+ */
+float Io_FlowMeters_GetPrimaryFlowRate(void);
+
+/**
+ * Get flow rate (L/min) from the secondary flow meter
+ */
+float Io_FlowMeters_GetSecondaryFlowRate(void);
+
+/**
+ * Updates the frequency for the flow meter's PWM output. This
+ * should be called in the input capture callback function.
+ * @param htim : The handle of the timer that triggered the input capture
+ * callback function
+ */
+void Io_FlowMeters_InputCaptureCallback(TIM_HandleTypeDef *htim);
+
+/**
+ * Detects when timer CNT register overflowed, and checks if positive edges are
+ * detected from the primary flow meter. Outputs NaN in situations where edges
+ * triggered interrupts are not detected.
+ */
+void Io_FlowMeters_Elapsed_TickPrimary(void);
```

Comment:
I found "elapsed" very confusing because it's really just an adjective used in the HAL driver. 
```suggestion
void Io_FlowMeters_CheckIfPrimaryIsActive(void);
```